File: solutions/java/hamming/1/src/main/java/Hamming.java

public class Hamming {

    private final String leftStrand;
    private final String rightStrand;

    public Hamming(String leftStrand, String rightStrand) {

        if (leftStrand.length() != rightStrand.length()) {
            throw new IllegalArgumentException("strands must be of equal length");
        }

        this.leftStrand = leftStrand.toLowerCase();
        this.rightStrand = rightStrand.toLowerCase();
    }

    public int getHammingDistance() {
        int distance = 0;

        for (int i = 0; i < leftStrand.length(); i++) {
            if (leftStrand.charAt(i) != rightStrand.charAt(i)) {
                distance++;
            }
        }

        return distance;
    }
}

File: solutions/kotlin/hamming/1/src/main/kotlin/Hamming.kt
object Hamming {

    fun compute(leftStrand: String, rightStrand: String): Int {
        if (leftStrand.length != rightStrand.length) {
            throw IllegalArgumentException("left and right strands must be of equal length")
        }
        var count: Int = 0;
        
        for ((index, char) in leftStrand.iterator().withIndex()) {
            if (char != rightStrand[index]) {
                count++;
            }
        }

        return count;
    }
}
